src/main/java/com/linagora/elasticsearch/metrics/ElasticsearchReporter.java

```diff
@@ -463,53 +532,67 @@ private HttpURLConnection openConnection(String uri, String method) {
-                json.writeEndObject();
-                json.writeEndObject();
-                json.writeEndObject();
-
-                json.writeEndObject();
-                json.writeEndObject();
-                json.flush();
-
-                putTemplateConnection.disconnect();
-                if (putTemplateConnection.getResponseCode() != 200) {
-                    LOGGER.error("Error adding metrics template to elasticsearch: {}/{}", putTemplateConnection.getResponseCode(), putTemplateConnection.getResponseMessage());
+                HttpPut putTemplateRequest = new HttpPut("/_template/metrics_template");
+                putTemplateRequest.setHeader(HttpHeaders.CONTENT_TYPE, "application/json");
+                putTemplateRequest.setHeader(HttpHeaders.CACHE_CONTROL, "no-cache");
+                byte[] requestJsonBody = writeIndexTemplateJson();
+                putTemplateRequest.setEntity(new ByteArrayEntity(requestJsonBody));
+
+                try (CloseableHttpResponse putTemplateResponse = executeRequest(putTemplateRequest)) {
+                    if (putTemplateResponse.getStatusLine().getStatusCode() != 200) {
+                        LOGGER.error("Error adding metrics template to elasticsearch: {}/{}",
+                            putTemplateResponse.getStatusLine().getStatusCode(), putTemplateResponse.getStatusLine().getReasonPhrase());
+                    }
+                    checkedForIndexTemplate = true;
+                } catch (FailedtoConnectToElasticSearchException e) {
+                    LOGGER.error("Error adding metrics template to elasticsearch", e);
                 }
+            } else {
+                checkedForIndexTemplate = true;
             }
-            checkedForIndexTemplate = true;
         } catch (IOException e) {
             LOGGER.error("Error when checking/adding metrics template to elasticsearch", e);
+        } catch (FailedtoConnectToElasticSearchException e) {
+            //Do nothing
+        }
+    }
+
+    private byte[] writeIndexTemplateJson() throws IOException {
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+
+        JsonGenerator json = new JsonFactory().createGenerator(outputStream);
+        json.writeStartObject();
+        json.writeStringField("template", index + "*");
+        json.writeObjectFieldStart("mappings");
+
+        json.writeObjectFieldStart("_default_");
+        json.writeObjectFieldStart("_all");
+        json.writeBooleanField("enabled", false);
+        json.writeEndObject();
+        json.writeObjectFieldStart("properties");
+        json.writeObjectFieldStart("name");
+        json.writeObjectField("type", "string");
+        json.writeObjectField("index", "not_analyzed");
+        json.writeEndObject();
+        json.writeEndObject();
+        json.writeEndObject();
+
+        json.writeEndObject();
+        json.writeEndObject();
+        json.flush();
+
+        return outputStream.toByteArray();
```

Comment:
why not just:
```java
String json = "{....}"
return json.toByteArray() ?
```

Reply:
I have just extracted the method, did not touch the content